src/hip_stream.cpp

```diff
@@ -254,14 +254,137 @@ hipError_t hipStreamGetPriority(hipStream_t stream, int* priority) {
 
 
 //---
+hipError_t setCallbackPacket(hsa_queue_t* queue,
+                             uint64_t& index, uint64_t& nextIndex,
+                             hsa_barrier_and_packet_t** barrier1,
+                             hsa_barrier_and_packet_t** barrier2){
+
+    if(queue == nullptr ||
```

Comment:
None of these checks are needed since setCallbackPacket() is an internal function and only called from within hipStreamAddCallback() where all the necessary checks have been performed.